File: php/Program/Baju.php
<?php
require_once 'Aksesoris.php';

class Baju extends Aksesoris {
    protected $untuk;
    protected $size;
    protected $merk;

    public function __construct($id, $name, $stock, $price, $photo, $jenis, $bahan, $warna, $untuk, $size, $merk) {
        parent::__construct($id, $name, $stock, $price, $photo, $jenis, $bahan, $warna);
        $this->untuk = $untuk;
        $this->size = $size;
        $this->merk = $merk;
    }

    public function setUntuk($untuk) { $this->untuk = $untuk; }
    public function getUntuk() { return $this->untuk; }
    
    public function setSize($size) { $this->size = $size; }
    public function getSize() { return $this->size; }
    
    public function setMerk($merk) { $this->merk = $merk; }
    public function getMerk() { return $this->merk; }
}

File: CPP/Program/Baju.cpp
#include <iostream>
#include <string>
#include "Aksesoris.cpp"

class Baju : public Aksesoris {
    private:
        std::string untuk;
        std::string size;
        std::string merk;
    
    public:
        // Konstruktor default
        Baju() : Aksesoris(), untuk(""), size(""), merk("") {}

        // Konstruktor dengan parameter (diperbaiki urutan & tipe harga)
        Baju(int ID, std::string nama, int stok_produk, double harga_produk, std::string jenis, std::string bahan, std::string warna, std::string untuk, std::string size, std::string merk) 
        : Aksesoris(ID, nama, stok_produk, harga_produk, jenis, bahan, warna), untuk(untuk), size(size), merk(merk) {}

        // Setter
        void set_untuk(std::string untuk) { this->untuk = untuk; }
        void set_size(std::string size) { this->size = size; }
        void set_merk(std::string merk) { this->merk = merk; }

        // Getter
        std::string get_untuk() const { return this->untuk; }
        std::string get_size() const { return this->size; }
        std::string get_merk() const { return this->merk; }

        // Destruktor
        ~Baju() {}
};
